### ISSUE TITLE
hit when double clicking a text file inside an archive

### ISSUE THREAD

**Issue by chris-y** [chris-y/avalanche#23]
- got a hit when douple clicking a text file inside an archive. Text file is openened correctly. See MuForce output below.

Best regards,

Marcus

11-Jul-22   18:05:21
BYTE READ from 00000000                        PC: 08ADFA5A
USP : 0A0304AC SR: 0004  (U0)(-)(-)  TCB: 0A02ED78
Data: 00000000 00000000 09E5C634 00000400 00000000 0A04BCEC 0A04C2D0 00000000
----> 0A04BCEC - "Avalanche"  Hunk 0000 Offset 0000101C
----> 0A04C2D0 - "Avalanche"  Hunk 0000 Offset 00001600
Addr: 00000000 ABADF00D 09E5C634 00000000 0A2245CC 0A030508 08014DB0 08002378
Stck: 09E5C634 08AE1244 00000000 00000400 0A030964 0A224560 00000000 0AFA6550
Stck: 0279718D 0A0513D6 0A224560 00000000 00000064 00001000 00000000 00000000
Stck: 00020003 0A030A02 0AFA6550 09FCD5AC 0A02CC5E 0800FC84 000001A0 52414D3A
Stck: 4176616C 616E6368 652E3631 2F617574 682F7376 6E2E7369 6D706C65 2F613532
----> 08ADFA5A - "LDM - dos 47.28 (18.10.2021)"  Hunk 0000 Offset 00000B72
----> 08AE1244 - "LDM - dos 47.28 (18.10.2021)"  Hunk 0000 Offset 0000235C
----> 0A0513D6 - "Avalanche"  Hunk 0000 Offset 00006706
PC-8: 2E402001 91C84E75 48E700C0 E5824EB9 08ADFA54 4CDF0300 4ED62F02 70002041
PC *: 4A186704 520066F8 53882202 E4892242 340043F1 00016002 132051C8 FFFC1302
08adfa4a :  08ad fa54 4cdf             bclr #$54,$4cdf(a5)
08adfa50 :  0300                       btst d1,d0
08adfa52 :  4ed6                       jmp (a6)
08adfa54 :  2f02                       move.l d2,-(a7)
08adfa56 :  7000                       moveq.l #$0,d0
08adfa58 :  2041                       movea.l d1,a0
08adfa5a : *4a18                       tst.b (a0)+
08adfa5c :  6704                       beq.s $8adfa62
08adfa5e :  5200                       addq.b #$1,d0
08adfa60 :  66f8                       bne.s $8adfa5a
08adfa62 :  5388                       subq.l #$1,a0
08adfa64 :  2202                       move.l d2,d1
08adfa66 :  e489                       lsr.l #$2,d1
08adfa68 :  2242                       movea.l d2,a1
Name: "Avalanche"

**Comment by chris-y** [comment]
`
21-Aug-22   17:26:09
BYTE READ from 00000000                        PC: 08AE121E
USP : 0BBEECFC SR: 0014  (U0)(-)(-)  TCB: 0BA7DBE0
Data: FFFFFFFF 0BBEED50 00000000 00000400 00000000 0BA6B1EC 0BA6B7DC 00000000
----> 0BA6B1EC - "Avalanche"  Hunk 0000 Offset 0000101C
----> 0BA6B7DC - "Avalanche"  Hunk 0000 Offset 0000160C
Addr: 00000000 ABADF00D 09DD03D8 00000000 09DD0444 0BBEED50 08014DB0 08002378
Stck: 00000000 00000400 0BBEF1AC 09DD03D8 00000000 0BA51C50 00000400 0BA70CCA
Stck: 09DD03D8 00000000 00000064 00001000 00000000 00000000 00020003 0BBEF256
Stck: 0BA51C50 0A15D31C 09D1DF20 0800FC84 00000114 543A4176 616C616E 6368652E
Stck: 35322F64 616E736B 2F537973 2F2E496D 61676573 2F626F69 6E672E69 66660090
----> 08AE121E - "LDM - dos 47.28 (18.10.2021)"  Hunk 0000 Offset 00002336
----> 0BA70CCA - "Avalanche"  Hunk 0000 Offset 00006AFA
PC-8: 243C0000 04046100 4946221F 4CDF203C 4E75594F 48E73134 26422A41 7E00204B
PC *: 4A1866FC 538891CB 20085280 6100E2F4 24404A80 6726200A E4802F40 0018220B
08ae120e :  4e75                       rts
08ae1210 :  594f                       subq.w #$4,a7
08ae1212 :  48e7 3134                  movem.l d2-d3/d7/a2-a3/a5,-(a7)
08ae1216 :  2642                       movea.l d2,a3
08ae1218 :  2a41                       movea.l d1,a5
08ae121a :  7e00                       moveq.l #$0,d7
08ae121c :  204b                       movea.l a3,a0
08ae121e : *4a18                       tst.b (a0)+
08ae1220 :  66fc                       bne.s $8ae121e
08ae1222 :  5388                       subq.l #$1,a0
08ae1224 :  91cb                       suba.l a3,a0
08ae1226 :  2008                       move.l a0,d0
08ae1228 :  5280                       addq.l #$1,d0
08ae122a :  6100 e2f4                  bsr $8adf520
Name: "Background CLI"  CLI: "Avalanche"  


21-Aug-22   17:26:10
BYTE READ from 00000000                        PC: 08ADFA5A
USP : 0BBEECF4 SR: 0004  (U0)(-)(-)  TCB: 0BA7DBE0
Data: 00000000 00000000 09D1F3E4 00000400 00000000 0BA6B1EC 0BA6B7DC 00000000
----> 0BA6B1EC - "Avalanche"  Hunk 0000 Offset 0000101C
----> 0BA6B7DC - "Avalanche"  Hunk 0000 Offset 0000160C
Addr: 00000000 ABADF00D 09D1F3E4 00000000 09DD0444 0BBEED50 08014DB0 08002378
Stck: 09D1F3E4 08AE1244 00000000 00000400 0BBEF1AC 09DD03D8 00000000 0BA51C50
Stck: 02747CF9 0BA70CCA 09DD03D8 00000000 00000064 00001000 00000000 00000000
Stck: 00020003 0BBEF256 0BA51C50 0A15D31C 09D1DF20 0800FC84 00000114 543A4176
Stck: 616C616E 6368652E 35322F64 616E736B 2F537973 2F2E496D 61676573 2F626F69
----> 08ADFA5A - "LDM - dos 47.28 (18.10.2021)"  Hunk 0000 Offset 00000B72
----> 08AE1244 - "LDM - dos 47.28 (18.10.2021)"  Hunk 0000 Offset 0000235C
----> 0BA70CCA - "Avalanche"  Hunk 0000 Offset 00006AFA
PC-8: 2E402001 91C84E75 48E700C0 E5824EB9 08ADFA54 4CDF0300 4ED62F02 70002041
PC *: 4A186704 520066F8 53882202 E4892242 340043F1 00016002 132051C8 FFFC1302
08adfa4a :  08ad fa54 4cdf             bclr #$54,$4cdf(a5)
08adfa50 :  0300                       btst d1,d0
08adfa52 :  4ed6                       jmp (a6)
08adfa54 :  2f02                       move.l d2,-(a7)
08adfa56 :  7000                       moveq.l #$0,d0
08adfa58 :  2041                       movea.l d1,a0
08adfa5a : *4a18                       tst.b (a0)+
08adfa5c :  6704                       beq.s $8adfa62
08adfa5e :  5200                       addq.b #$1,d0
08adfa60 :  66f8                       bne.s $8adfa5a
08adfa62 :  5388                       subq.l #$1,a0
08adfa64 :  2202                       move.l d2,d1
08adfa66 :  e489                       lsr.l #$2,d1
08adfa68 :  2242                       movea.l d2,a1
Name: "Background CLI"  CLI: "Avalanche"  

`

**Comment by chris-y** [comment]
```
7.AmigaOS:> findhunkoffset "RAM Disk:Avalanche" 0x0000 0x0101C
Section 0, offset 0x0000101c is at
_fuelgauge_update + 0x00000330
in NAS:git-repos/Avalanche/src/avalanche.c line 551.
7.AmigaOS:> findhunkoffset "RAM Disk:Avalanche" 0x0000 0x0160C
Section 0, offset 0x0000160c is at
_fuelgauge_update + 0x00000920
in NAS:git-repos/Avalanche/src/avalanche.c line 1.
```

**Comment by chris-y** [comment]
Suspect some memory is getting overwritten as these offsets make no sense.

**Comment by chris-y** [comment]
Closing this as suspect it was fixed in #48
